ribs/emitters/opt/_gradient_opt_base.py

```diff
@@ -0,0 +1,38 @@
+"""Provides GradientOptBase."""
+from abc import ABC, abstractmethod
+
+
+class GradientOptBase(ABC):
```

Comment:
We might be able to add a constructor to initialize step_size? All gradient opt needs it right?